services/workflows-service/src/document/document.service.ts

```diff
@@ -1100,14 +1210,16 @@ export class DocumentService {
 
   getLatestDocumentVersions(documents: Document[]) {
     const documentsByType = documents.reduce((acc, document) => {
-      const documentId = getDocumentId(
-        {
-          type: document.type,
-          category: document.category,
-          issuingCountry: document.issuingCountry,
-        },
-        false,
-      );
+      const documentId = document.businessId
+        ? getDocumentId(
+            {
+              type: document.type,
+              category: document.category,
+              issuingCountry: document.issuingCountry,
+            },
+            false,
+          )
+        : `${document.endUserId}-${document.type}-${document.category}-${document.issuingCountry}`;
```

Comment:
Maybe there is a better condition to decide when to use the id with the entity id?